### DISCUSSION TITLE
Validate JSON type

### DISCUSSION THREAD

**Discussion by roylisto** [cue-lang/cue#1202] [Q&A] [answered]
What type to use to validate the value is in JSON format or not based on this doc?
https://cuelang.org/docs/tutorials/tour/types/types/

**Comment by myitcv** [comment]
@roylisto please can you provide an example of what it is you are trying to do? I think I follow, but would like to confirm by way of a concrete example.

**Reply by roylisto** [reply]
so I have a key that contains JSON value, I just want to validate if the value is in JSON type, can I use struct for that?
for example
metadata: JSON

**Reply by myitcv** [reply]
See @verdverm's answer below.

**Comment by verdverm** [comment, marked as answer]
```
import "encoding/json"

metadata: json.Valid
```

https://pkg.go.dev/cuelang.org/go@v0.4.0/pkg/encoding/json#Valid

https://cuelang.org/play/?id=e8dWbmXkGxM#cue@export@cue

There is also a `Validate` helper if you want to supply the schema as well

https://cuelang.org/play/?id=3kqDkvRlmT8#cue@export@cue

You can change the data or schema to see how these example will error with invalid input
